autoload/codefmt/ruff.vim

```diff
@@ -19,7 +19,7 @@ let s:plugin = maktaba#plugin#Get('codefmt')
 function! s:FormatWithArgs(args) abort
   let l:executable = s:plugin.Flag('ruff_executable')
   let l:lines = getline(1, line('$'))
-  let l:cmd = [l:executable, 'format'] + a:args
+  let l:cmd = [l:executable, 'format'] + a:args + ['-']
```

Comment:
This should be in the else block, the commands should be:

When the filename is available:

```
ruff format <args> --stdin-filename <filename>
```

When filename is not available

```
ruff format <args> --stdin-filename <placeholder>
```

Where `<placeholder>` can be `-`, or (I would suggest something like `currentbuffer`

Reply:
Hmm, there might be a misunderstanding about how the args work. The intended behavior is:
- If there's a known filename: `... <args> - --stdin-filename=foo.py`
- Otherwise: `... <args> -`

Either way, it should have the special `-` placeholder to signal that it should read from stdin, not try to format files in-place, and the `--stdin-filename` value is just an optional hint of the filepath if available to help the formatter understand what it's formatting (for instance, if the project ruff config had per-path overrides).

Reply:
According to the current source, no:
https://github.com/astral-sh/ruff/blob/main/crates/ruff/src/lib.rs#L87-L104

But feel free to leave as is.

Reply:
Isn't that just saying the explicit `-` is optional but not redundant in that case?

If `- --stdin-filename=foo.py` would tend to confuse ruff, then +1 I'd like to change it to drop the `-` in that case, but otherwise I think it expresses the intention better to say we always have a STDIN `-` and sometimes have a `--stdin-filename`.